Core/URLExtension.swift

```diff
@@ -141,7 +141,7 @@ extension URL {
         }
 
         // from https://stackoverflow.com/a/25717506/73479
-        let hostNameRegex = "^(((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,63})$"
+        let hostNameRegex = "^(((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z0-9-]{2,63})$"
```

Comment:
The change allows hyphens in the domain name which is what happens when characters are punycoded. In actual fact this regex is slightly wrong because it doesn't allow local host names but that suits has as we treat them like a search.